azkaban-common/src/main/java/azkaban/utils/PatternLayoutEscaped.java

```diff
@@ -0,0 +1,39 @@
+package azkaban.utils;
+
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+
+/**
+ * When we use log4j to send a JSON message the official PatternLayout will send corrupt JSON if certain characters are
+ * present. We use this PatternLayoutEscaped class as a thin interface around PatternLayout to escape all these
+ * characters.
+ */
+public class PatternLayoutEscaped extends PatternLayout {
```

Comment:
Why slf4j? Looking at the source for log4j2's PatternLayout class these will not be compatible. Also I believe that with log4j2, a scheme like this is not necessary.